src/form/en_ivr.jl

```diff
@@ -0,0 +1,1763 @@
+		pm::AbstractExplicitNeutralIVRModel,
+		nw::Int,
+		f_idx::Tuple{Int,Int,Int},
+		connections::Vector{Int},
+		rating::Vector{<:Real}
+	)
+
+For IVR models with explicit neutrals, 
+imposes a bound on the switch current magnitude per conductor.
+Note that a bound on the from-side implies the same bound on the to-side current, 
+so it suffices to apply this only explicitly at the from-side.
+"""
+function constraint_mc_switch_current_limit(pm::AbstractExplicitNeutralIVRModel, nw::Int, f_idx::Tuple{Int,Int,Int}, connections::Vector{Int}, rating::Vector{<:Real})
+    crsw = var(pm, nw, :crsw, f_idx)
+    cisw = var(pm, nw, :cisw, f_idx)
+
+    for idx in 1:length(rating)
+        if rating[idx] < Inf
+            JuMP.@constraint(pm.model, crsw[idx]^2 + cisw[idx]^2 <= rating[idx]^2)
+        end
+    end
+end
+
+
+"""
+	function constraint_mc_switch_thermal_limit(
+		pm::AbstractNLExplicitNeutralIVRModel,
+		nw::Int,
+		f_idx::Tuple{Int,Int,Int},
+		f_connections::Vector{Int},
+		rating::Vector{<:Real}
+	)
+
+For IVR models with explicit neutrals, 
+imposes a bound on the switch power magnitude per conductor.
+Note that a bound on the from-side implies the same bound on the to-side power 
+when the switch is closed (equal voltages), and also when it is open since the 
+power then equals zero on both ends.
+"""
+function constraint_mc_switch_thermal_limit(pm::AbstractNLExplicitNeutralIVRModel, nw::Int, f_idx::Tuple{Int,Int,Int}, f_connections::Vector{Int}, rating::Vector{<:Real})
+    vr_fr = [var(pm, nw, :vr, f_idx[1])[t] for t in f_connections]
+    vi_fr = [var(pm, nw, :vi, f_idx[1])[t] for t in f_connections]
+    crsw_fr = var(pm, nw, :crsw, f_idx)
+    cisw_fr = var(pm, nw, :cisw, f_idx)
+
+    for idx in 1:length(rating)
+        if rating[idx] < Inf
+            psw_fr_idx = JuMP.@NLexpression(pm.model,  vr_fr[idx]*crsw_fr[idx] + vi_fr[idx]*cisw_fr[idx])
+            qsw_fr_idx = JuMP.@NLexpression(pm.model, -vr_fr[idx]*cisw_fr[idx] + vi_fr[idx]*crsw_fr[idx])
+
+            JuMP.@NLconstraint(pm.model, psw_fr_idx^2 + qsw_fr_idx^2 <= rating[idx]^2)
+        end
+    end
+end
+
+
+"""
+	function constraint_mc_switch_thermal_limit(
+		pm::AbstractQuadraticExplicitNeutralIVRModel,
+		nw::Int,
+		f_idx::Tuple{Int,Int,Int},
+		f_connections::Vector{Int},
+		rating::Vector{<:Real}
+	)
+
+For quadratic IVR models with explicit neutrals, 
+throw an error because this cannot be represented quadratically 
+without introducing explicit power variables.
+"""
+function constraint_mc_switch_thermal_limit(pm::AbstractQuadraticExplicitNeutralIVRModel, nw::Int, f_idx::Tuple{Int,Int,Int}, f_connections::Vector{Int}, rating::Vector{<:Real})
+    if any(rating.<Inf)
+        error("""
+            A switch power bound cannot be represented quadratically in the default AbstractQuadraticExplicitNeutralIVRModel.
+            Either extend this quadratic formulation by including explicit switch power variables, or use AbstractNLExplicitNeutralIVRModel instead.""")
```

Comment:
Should this be an error? I think it might be better to just be a warning and to ignore the thermal limit, which is more in-line with current behavior elsewhere in the package